core/src/main/java/org/elasticsearch/index/query/IndicesQueryParser.java

```diff
@@ -145,6 +148,7 @@ public Query parse(QueryParseContext parseContext) throws IOException, QueryPars
         if (queryName != null) {
             parseContext.addNamedQuery(queryName, chosenQuery);
         }
+        chosenQuery.setBoost(boost);
```

Comment:
Maybe move this up before the if? Shouldn't make a difference as long as addNamedQuery doesn't copy, but just to show that chosenQuery is "complete" when adding it to the named queries.